### PR TITLE
fix(microsoft-foundry): conform tests and sanitize credentials to prevent Unicode encoding errors

### PR DESCRIPTION
**Pull request by yesudeep**
## Summary

Credentials (API keys, endpoint URLs, API versions) copied from web UIs
like Azure Portal often contain invisible Unicode characters such as
zero-width spaces (U+200B). These propagate through the OpenAI SDK's
request chain and cause `UnicodeEncodeError: 'ascii' codec can't encode
character '\u200b'` failures deep inside the HTTP transport layer,
particularly on Linux systems with ASCII-defaulting locales.

This PR also fixes two usability issues reported by users setting up
Genkit samples on fresh Linux VMs (GCE Debian): the `genkit` CLI not
being found, and `uv` not being on PATH after running `setup.sh`.

Fixes: https://github.com/firebase/genkit/issues/4688
Fixes: https://github.com/firebase/genkit/issues/4684

### Credential sanitization (`microsoft-foundry` plugin)

- Add `_sanitize_credential()` with precompiled regex for 10 categories
  of invisible Unicode characters (zero-width spaces, BOMs, non-breaking
  spaces, directional marks, word joiners, line/paragraph separators)
- Apply sanitization to `api_key`, `endpoint`, and `api_version` in
  `MicrosoftFoundry.__init__`
- Add `TestSanitizeCredential` test class with 9 tests covering None
  passthrough, individual character types, combined stripping, and an
  integration test verifying env var sanitization during init

### Sample scripts (`_common.sh`, `setup.sh`)

- **Auto-discover tool paths**: Add `_ensure_tool_paths()` that sources
  `~/.environment` and adds common install directories (`~/.local/bin`,
  `~/.local/share/pnpm`, `~/.cargo/bin`, npm global prefix) to PATH.
  This fixes `uv: command not found` when tools were installed by
  `setup.sh` but the user has not sourced `~/.environment` yet.
- **Pre-flight check for genkit CLI**: `genkit_start_with_browser()` now
  detects a missing `genkit` command and interactively offers to install
  pnpm + genkit-cli, instead of the cryptic `stdbuf: failed to run
  command 'genkit': No such file or directory` error.
- **Prefer package manager for Ollama on Linux**: `_install_ollama()` in
  both `setup.sh` and `_common.sh` now tries `apt`/`dnf` before falling
  back to the curl installer, matching the pattern used by other tools.

### CODE REVIEW

File: py/samples/_common.sh [hunk 1]
```diff
@@ -196,6 +196,66 @@ _watch_for_devui_url() {
...
+                    pnpm install -g genkit-cli
+                fi
+            elif command -v npm &> /dev/null; then
+                echo -en "${BLUE}Install genkit CLI via npm? [Y/n]:${NC} "
+                local response
+                read -r response < /dev/tty
+                if [[ -z "$response" || "$response" =~ ^[Yy] ]]; then
+                    npm install -g genkit-cli
+                fi
+            fi
```
[the remark this answers is not in the document]
Comment (yesudeep) [reply py/samples/_common.sh:231]: Good ssuggestion. Done.
